src/layout/layout.rs

```diff
@@ -889,12 +1149,138 @@ impl Element {
     }
 }
 
+#[derive(Debug)]
+struct StrengthSet {
+    pub spacer_size_equality: f64,
+    pub proportional_scaling_equality: f64,
+    pub fixed_value_equality: f64,
+    pub min_max_inequality: f64,
+    pub length_equality: f64,
+    pub percentage_equality: f64,
+    pub ratio_equality: f64,
+    pub min_max_equality: f64,
+    pub proportional_grower: f64,
+    pub grower: f64,
+    pub space_grower: f64,
+}
+
+impl StrengthSet {
+    pub fn is_valid(&self) -> bool {
+        self.spacer_size_equality > self.fixed_value_equality
+            && self.proportional_scaling_equality > self.fixed_value_equality
+            && self.fixed_value_equality > self.min_max_inequality
+            && self.min_max_inequality > self.min_max_equality
+            && self.min_max_inequality > self.length_equality
+            && self.length_equality > self.percentage_equality
+            && self.percentage_equality > self.ratio_equality
+            && self.ratio_equality > self.min_max_equality
+            && self.min_max_inequality > self.proportional_grower
+            && self.proportional_grower > self.grower
+            && self.grower > self.space_grower
+    }
+}
+
+impl Default for StrengthSet {
+    fn default() -> Self {
+        let (
```

Comment:
I think this method should probably just be 13 LoC (if you keep it as a struct and not a module):
```rust
Self {
    spacer_size_equality: REQUIRED - 1,
    proportional_scaling_equality: REQUIRED - 1,
    fixed_value_equality: REQUIRED / 10.0,
    min_max_inequality: STRONG * 10.0
    length_equality: STRONG / 10.0
    percentage_equality: MEDIUM * 10.0,  
    ratio_equality: MEDIUM,
    min_max_equality: MEDIUM / 10.0,
    grower: WEAK * 10.0,
    space_grower: WEAK,
    proportional_grower: WEAK / 10.0,
}
```